lib/sharezone_widgets/lib/src/launch_markdown_link_with_warning.dart

```diff
@@ -0,0 +1,295 @@
+                          dontWarnAgainForDomain: false,
+                        ),
+                      ),
+                  child: Text('Abbrechen', style: theme.textTheme.labelLarge),
+                ),
+                FilledButton(
+                  onPressed:
+                      () => Navigator.of(builderContext).pop(
+                        _LinkDialogResult(
+                          open: true,
+                          dontWarnAgainForDomain: dontWarnAgainForDomain,
+                        ),
+                      ),
+                  child: const Text('Link öffnen'),
+                ),
+              ],
+            );
+          },
+        ),
+      );
+    },
+  );
+}
+
+bool _isDomainTrusted({
+  required KeyValueStore keyValueStore,
+  required String domain,
+}) {
+  final List<String>? stored = keyValueStore.tryGetStringList(
+    trustedDomainsStoreKey,
+  );
+
+  if (stored == null || stored.isEmpty) {
+    return false;
+  }
+
+  return stored.any((value) => value.toLowerCase() == domain.toLowerCase());
+}
+
+Future<void> _storeTrustedDomain({
+  required KeyValueStore keyValueStore,
+  required String domain,
+}) async {
+  final List<String> current = List<String>.from(
+    keyValueStore.tryGetStringList(trustedDomainsStoreKey) ?? [],
+  );
+
+  final String normalized = domain.toLowerCase();
+  if (current.any((value) => value.toLowerCase() == normalized)) {
+    return;
+  }
+
+  current.add(normalized);
+  current.sort();
+
+  await keyValueStore.setStringList(trustedDomainsStoreKey, current);
+}
+
+/// Tries to parse [input] as a URI.
+///
+/// Adds a `https://` prefix if [input] looks like a domain (e.g.,
+/// `example.com` or `www.example.com`).
+///
+/// Returns `null` if [input] is empty or cannot be parsed as a URI.
+@visibleForTesting
+Uri? toLaunchableUri(String input) {
+  final trimmed = input.trim();
+
+  if (trimmed.isEmpty) return null;
+
+  final Uri? initial = Uri.tryParse(trimmed);
+  if (initial == null) return null;
+
+  // Accept URIs that already have a scheme (e.g., https, http, mailto, tel)
+  // Only treat as a proper scheme if the original string contains '://' to
+  // avoid misclassifying inputs like 'example.com:8080' where Dart may parse
+  // 'example.com' as a scheme.
+  if (initial.hasScheme && trimmed.contains('://')) {
+    return initial;
+  }
+
+  // If there is no scheme and it looks like a domain (e.g., example.com or
+  // www.example.com), optionally with a port, assume https and try again.
+  final bool looksLikeDomain =
+      RegExp(
+        r'^[A-Za-z0-9.-]+\.[A-Za-z]{2,}(:[0-9]+)?(/.*)?$',
+      ).hasMatch(trimmed) ||
+      trimmed.startsWith('www.');
+
+  if (looksLikeDomain) {
+    final Uri? httpsUri = Uri.tryParse('https://$trimmed');
+    if (httpsUri != null) {
+      return httpsUri;
+    }
+  }
+
+  // Fallback to the initial parse result (may be a relative path which we won't
+  // launch)
+  return initial;
+}
```

Comment:
This method makes me very uneasy xD

Lets hope that this will not be a place for a future vulnerability.